Services/SSLCertificateService.cs

```diff
@@ -187,6 +188,7 @@ public X509Certificate2 GetCertificate(StoreLocation storeLocation, StoreName st
 
         public X509Certificate2 GetCertificate(StoreLocation storeLocation, StoreName storeName, string certThumbPrint,  string friendlyName, string certSubjectName, string dnsName, Nullable<DateTime> notAfter)
         {
+            var result = new List<X509Certificate2>();
```

Comment:
validCertificates, availableCertificates, certificatesMatchingNotNullParameters, instead of result ?